File: frontend/src/types/payment.ts
export type PaymentStatus = "completed" | "processing" | "failed";

export interface Payment {
  id: string;
  merchant: string;
  status: PaymentStatus;
  amount: string;
  created_at: string;
}

export interface PaymentListParams {
  status?: string;
  id?: string;
  sort?: string;
}

export interface PaymentListResponse {
  payments: Payment[];
}

File: backend/internal/entity/payment.go
package entity

import "time"

type PaymentStatus string

const (
	PaymentStatusCompleted   PaymentStatus = "completed"
	PaymentStatusProcessing  PaymentStatus = "processing"
	PaymentStatusFailed      PaymentStatus = "failed"
)

type Payment struct {
	ID        string        `json:"id"`
	Merchant  string        `json:"merchant"`
	Status    PaymentStatus `json:"status"`
	Amount    string        `json:"amount"`
	CreatedAt time.Time     `json:"created_at"`
}
